### PR TITLE
add validation to user-ids-by-hostnames handler

### PR DESCRIPTION
**Pull request by ddellacosta**

### CODE REVIEW

**Comment by ddellacosta** [comment]
@cap10morgan Sorry I should have added this to the last one--would you mind taking a look?

File: src/user_works/handlers.clj [hunk 1]
```diff
@@ -213,8 +213,12 @@
      :error {:type :not-found}
      :id id}))
 
+(def user-ids-by-hostnames-schema
+  {:hostnames (s/cond-pre (s/enum :all) [s/Str])})
```
Comment (cap10morgan) [inline src/user_works/handlers.clj]: Sorry I didn't catch this before. Yes, schemas are good.

Also: should hostnames be a vector or a set? Does order of the hostnames in the request matter?

File: src/user_works/handlers.clj [hunk 2]
```diff
@@ -213,8 +213,12 @@
...
+
 (defn user-ids-by-hostnames
   [{:keys [hostnames] :as message}]
```
Comment (cap10morgan) [inline src/user_works/handlers.clj]: Can we spit out a debug-level log message along the lines of `"Received user-ids-by-hostnames request:" (pr-str message)` here?

File: test/user_works/handlers_test.clj [hunk 3]
```diff
@@ -835,4 +835,14 @@
...
+
+    (testing "validates incoming message"
+      (is (thrown? java.lang.RuntimeException
+                   (handlers/user-ids-by-hostnames {:hostnames "foo"})))
+      (is (thrown? java.lang.RuntimeException
+                   (handlers/user-ids-by-hostnames {:hostnames [:foo]})))
+      (is (thrown? java.lang.RuntimeException
+                   (handlers/user-ids-by-hostnames {:hostnames :not-all})))
+      (is (thrown? java.lang.RuntimeException
+                   (handlers/user-ids-by-hostnames {}))))))
```
Comment (cap10morgan) [inline test/user_works/handlers_test.clj]: Should we use `thrown-with-msg?` here and verify that it's a schema validation exception?

File: src/user_works/handlers.clj (hunk 1 above)
Comment (ddellacosta) [inline src/user_works/handlers.clj]: Order doesn't matter, can and probably should be a set.

**Comment by ddellacosta** [comment]
@cap10morgan Take another look?

**Comment by cap10morgan** [comment]
LGTM

**Pull request merged**